compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -1081,6 +1121,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             }
         }
 
+        // Incorporate the argument changes in the removal suggestion.
+        let mut prev = -1;
+        for (expected_idx, provided_idx) in matched_inputs.iter_enumerated() {
+            if let Some(provided_idx) = provided_idx {
+                prev = provided_idx.index() as i64;
+            }
+            let idx = ProvidedIdx::from_usize((prev + 1) as usize);
+            if let None = provided_idx
+                && let Some((_, arg_span)) = provided_arg_tys.get(idx)
+            {
+                let (_, expected_ty) = formal_and_expected_inputs[expected_idx];
+                suggestions.push((*arg_span, ty_to_snippet(expected_ty, expected_idx)));
+            }
+        }
```

Comment:
What do you mean by  "breaking your code"? Do you have an example of a broken suggestion that we can put as a test?

estebank: I don't really understand what this is doing either.

Reply:
It's been a while since I wrote this and now I don't recall *exactly* what this does, but this weird code was so that I could also incorporate the "change this argument for this other one" suggestions into the removal one, but it is quite messy :-/

Reply:
The last commit now shows the effect of this block taken out on the tests.